components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/SequenceGenerator.java

```diff
@@ -0,0 +1,377 @@
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.registry.core.RegistryConstants;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.wso2.carbon.apimgt.impl.utils.APIUtil.handleException;
+
+/**
+ * Class uses to generate api sequences for soap to rest conversion
+ */
+public class SequenceGenerator {
+
+    private static APIDefinition definitionFromSwagger20 = new APIDefinitionFromSwagger20();
+
+    /**
+     * generates api in/out sequences from the swagger to soap definitions
+     *
+     * Note: this method is directly invoked from the jaggery layer
+     *
+     * @param apiDataStr api object as a string
+     * @param soapOperationMapping soap operation mapping from the wsdl
+     * @throws APIManagementException throws if error occurred in registry access/ parsing definitions
+     */
+    public static void generateSequences(String apiDataStr, String soapOperationMapping) throws APIManagementException {
+        JSONParser parser = new JSONParser();
+        boolean isTenantFlowStarted = false;
+        try {
+            JSONObject apiData = (JSONObject) parser.parse(apiDataStr);
+            String provider = (String) apiData.get("provider");
+            String name = (String) apiData.get("name");
+            String version = (String) apiData.get("version");
+
+            if (provider != null) {
+                provider = APIUtil.replaceEmailDomain(provider);
+            }
+
+            provider = (provider != null ? provider.trim() : null);
+            name = (name != null ? name.trim() : null);
+            version = (version != null ? version.trim() : null);
+            APIIdentifier apiId = new APIIdentifier(provider, name, version);
+
+            JSONObject apiJSON;
+
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(provider));
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
```

Comment:
We need to start the tenant flow inside the try block, if there is an exception happens before going to try block, tenent flow will not end.

Reply:
it's handled by the try block at https://github.com/wso2/carbon-apimgt/pull/4997/files/49f6c35434f4e89cf1a39d616ebd4ddce45cbc5a#diff-428fd425b9d5b7805d6a0db773295492R89